### PR TITLE
Rm urllib

### PR DESCRIPTION
**Pull request by deepaklorkhatri007**
This pull request addresses issue #...

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
Cross reference issue #2420

**Comment by deepaklorkhatri007** [comment]
@peterjc what is the cross-reference issue?

**Comment by deepaklorkhatri007** [comment]
oh i got it okay first i thought code has some issue, sorry for bothering you.

**Comment by peterjc** [comment]
On the pull request template, I expected you to fill in this line:

> This pull request addresses issue #...

to read:

> This pull request addresses issue #2420 

I could have edited your pull request description, but added the issue number in a comment instead. Either way we get an automatic link shown on the issue to this pull request, so anyone looking there is aware of this work.

Can you suggest how better to reword the template to better guide people?

**Comment by deepaklorkhatri007** [comment]
First of all, I think it is my lack of attention to things that led to this mistake and when I submitted the pull request I should have read every bit carefully but now I know.

> Can you suggest how better to reword the template to better guide people?

The pull request template is more than fine and I guess instead of `Cross reference issue #1234` you can write the same thing people have missed i.e `This pull request addresses issue #1234` with a message reminding of the mistake they have done. I checked and I missed it in the last pull request too, if I have got a comment on that I have not done the mistake again. I will try to be more attentive while doing things. Thank you so much for making me a better contributor.

**Comment by peterjc** [comment]
I'll leave this open another day or so in case anyone else wants to review it.

There's a trivial merge conflict, and assuming we merge #2478 next there will be another one.

**Comment by peterjc** [comment]
Squashed and rebased...

File: Bio/Blast/NCBIWWW.py [hunk 1]
```diff
@@ -19,9 +19,9 @@
 
 from io import StringIO
 from Bio._py3k import _as_string, _as_bytes
-from Bio._py3k import urlopen as _urlopen
-from Bio._py3k import urlencode as _urlencode
-from Bio._py3k import Request as _Request
+from urllib.request import urlopen as _urlopen
```
Comment (mdehoon) [inline Bio/Blast/NCBIWWW.py]: Why the underscores?

File: Bio/Entrez/Parser.py [hunk 2]
```diff
@@ -45,8 +45,8 @@
 from xml.sax.saxutils import escape
 
 # Importing these functions with leading underscore as not intended for reuse
-from Bio._py3k import urlopen as _urlopen
-from Bio._py3k import urlparse as _urlparse
+from urllib.request import urlopen as _urlopen
```
Comment (mdehoon) [inline Bio/Entrez/Parser.py]: Bio.Entrez has already been fixed.

File: Bio/Entrez/__init__.py [hunk 3]
```diff
@@ -113,9 +113,9 @@
 import warnings
 
 # Importing these functions with leading underscore as not intended for reuse
-from Bio._py3k import urlopen as _urlopen
```
Comment (mdehoon) [inline Bio/Entrez/__init__.py]: Bio.Entrez has already been fixed

File: Bio/ExPASy/ScanProsite.py [hunk 4]
```diff
@@ -6,8 +6,8 @@
 """Code for calling and parsing ScanProsite from ExPASy."""
 
 # Importing these functions with leading underscore as not intended for reuse
-from Bio._py3k import urlopen as _urlopen
-from Bio._py3k import urlencode as _urlencode
+from urllib.request import urlopen as _urlopen
```
Comment (mdehoon) [inline Bio/ExPASy/ScanProsite.py]: Same here: Why the underscores?

File: Bio/ExPASy/__init__.py [hunk 5]
```diff
@@ -17,7 +17,7 @@
 """
 
 # Importing these functions with leading underscore as not intended for reuse
-from Bio._py3k import urlopen as _urlopen
+from urllib.request import urlopen as _urlopen
```
Comment (mdehoon) [inline Bio/ExPASy/__init__.py]: same here

File: Bio/Graphics/KGML_vis.py [hunk 6]
```diff
@@ -32,7 +32,7 @@
 
     raise MissingPythonDependencyError("Install pillow if you want to use KGML_vis.")
 
-from Bio._py3k import urlopen as _urlopen
+from urllib.request import urlopen as _urlopen
```
Comment (mdehoon) [inline Bio/Graphics/KGML_vis.py]: underscores can be dropped in this files and all others

File: Tests/run_tests.py [hunk 7]
```diff
@@ -232,12 +232,12 @@ def main(argv):
...
             requires_internet.check.available = False
             # Monkey patch for urlopen()
-            import Bio._py3k
+            import urllib
 
             def dummy_urlopen(url):
                 raise RuntimeError("Internal test suite error, attempting to use internet despite --offline setting")
 
-            Bio._py3k.urlopen = dummy_urlopen
+            urllib.urlopen = dummy_urlopen
```
Comment (mdehoon) [inline Tests/run_tests.py]: Do we really need this?

Comment (peterjc) [reply Tests/run_tests.py]: Yes - it is to make sure ``--offline`` is actually working.

File: Bio/Graphics/KGML_vis.py (hunk 6 above)
Comment (peterjc) [reply Bio/Graphics/KGML_vis.py]: Not relevant to the py3k removal, but perhaps worth discussing on a separate issue?

File: Bio/Entrez/__init__.py (hunk 3 above)
Comment (peterjc) [reply Bio/Entrez/__init__.py]: It hadn't at the time I rebased this - but that explains the new merge conflict.

File: Bio/Blast/NCBIWWW.py (hunk 1 above)
Comment (deepaklorkhatri007) [reply Bio/Blast/NCBIWWW.py]: i thought of removing them but i wanted to consult here before doing that.

File: Bio/Graphics/KGML_vis.py (hunk 6 above)
Comment (deepaklorkhatri007) [reply Bio/Graphics/KGML_vis.py]: I am fine with a separate issue for resolving underscore removal.

File: Bio/Entrez/__init__.py (hunk 3 above)
Comment (peterjc) [reply Bio/Entrez/__init__.py]: Rebasing...

File: Bio/Entrez/Parser.py (hunk 2 above)
Comment (peterjc) [reply Bio/Entrez/Parser.py]: Rebased accordingly

File: Bio/ExPASy/ScanProsite.py (hunk 4 above)
Comment (peterjc) [reply Bio/ExPASy/ScanProsite.py]: Let's discuss this on #2483

File: Bio/ExPASy/__init__.py (hunk 5 above)
Comment (peterjc) [reply Bio/ExPASy/__init__.py]: same reply, let's discuss this on #2483

File: Bio/Graphics/KGML_vis.py (hunk 6 above)
Comment (peterjc) [reply Bio/Graphics/KGML_vis.py]: Let's discuss this on #2483

File: Bio/Blast/NCBIWWW.py (hunk 1 above)
Comment (peterjc) [reply Bio/Blast/NCBIWWW.py]: Let's discuss this on #2483

File: Bio/ExPASy/ScanProsite.py (hunk 4 above)
Comment (peterjc) [reply Bio/ExPASy/ScanProsite.py]: Let's discuss this on #2483

File: Bio/ExPASy/__init__.py (hunk 5 above)
Comment (peterjc) [reply Bio/ExPASy/__init__.py]: Let's discuss this on #2483

File: Bio/Graphics/KGML_vis.py (hunk 6 above)
Comment (peterjc) [reply Bio/Graphics/KGML_vis.py]: Let's discuss this on #2483

**Comment by peterjc** [comment]
@mdehoon with the Entrez conflicts done, are you in favour of merging this now?

**Comment by mdehoon** [comment]
As the underscores were added as part of _py3k, they should be addressed as part of the removal of _py3k.

**Comment by peterjc** [comment]
@mdehoon Since this contributed pull request doesn't change the underscore status quo, and just does what I was suggesting in #2420, and we merge it now please?

And yes, these particular leading underscores were added as part of the introduction of ``Bio._py3k`` (mostly if not all by me, e.g. fa6fe3ab2f38925e62fa615a856be8686f4a1b3a), but as I described on #2483 it was deliberate, and to me this is a quite separate issue.

**Comment by mdehoon** [comment]
@peterjc  What I meant was that because the underscores were added as part of _py3k as a temporary measure, and were not discussed or agreed upon to be a permanent feature of Biopython, the underscores should be removed in this PR. Then in #2483 we can discuss as a separate issue whether underscores should be added to all local imports.

**Comment by peterjc** [comment]
@mdehoon @deepaklorkhatri007 OK then, remove the underscores on the lines changed in this pull request.

I still hope to convince you on #2483 to put them all back again, but I don't seem to be doing a good job articulating my concern.

**Comment by deepaklorkhatri007** [comment]
>I still hope to convince you on #2483 to put them all back again, but I don't seem to be doing a good job articulating my concern.

@peterjc if the imports without underscores create any problems in the future, I am happy to add them back to their respective places.

**Comment by peterjc** [comment]
Michiel made a good point on #2483 about the Python standard library not being consistent on this point either, so I'm going to give up on the underscores here. As he says, there are more pressing things to work on.

**Comment by peterjc** [comment]
Merged, thank you!

**Pull request merged**